### PR TITLE
Add API key template

### PR DESCRIPTION
**Pull request by manlinl**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: mixer/template/apikey/template.proto [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+syntax = "proto3";
+
+package apiKey;
+
+import "google/protobuf/timestamp.proto";
+import "mixer/v1/template/extensions.proto";
+
+option (istio.mixer.v1.template.template_variety) = TEMPLATE_VARIETY_CHECK;
+
+// Template to check if an API call should proceed.
```
Comment (douglas-reid) [inline mixer/template/apikey/template.proto]: Would you mind adding an example config using this Template?

File: mixer/template/apikey/template.proto [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    string api_version = 2;
+
+    // The API operation is being called.
+    string api_operation = 3;
+
+    // API key used in API call.
+    string api_key = 4;
+
+    // Timestamp of API call.
+    google.protobuf.Timestamp timestamp = 5;
```
Comment (douglas-reid) [inline mixer/template/apikey/template.proto]: should we name this `request_time` or `operation_time` or `call_time` to match the style of the standard fields in https://cloud.google.com/apis/design/standard_fields ?

Comment (manlinl) [reply mixer/template/apikey/template.proto]: IMHO, in this context, it doesn't appear to be absolutely necessary to be as precise as call_time, just like the timestamp field in logentry template.

File: mixer/template/apikey/template.proto (hunk 1 above)
Comment (manlinl) [reply mixer/template/apikey/template.proto]: done.

**Comment by manlinl** [comment]
comment addressed, PTAL.

**Comment by manlinl** [comment]
@douglas-reid Hey Douglas, can yo take a look of the updated PR pls?

**Pull request merged**